docs/docs/guide/08-loan/06-liquidate.md

```diff
@@ -0,0 +1,87 @@
+# Liquidate loan
+
+The `LiquidateLoan` method is a function that allows the lender to sell off the
+collateral belonging to the borrower in the event that the borrower has failed
+to repay the loan by the specified deadline. This process is known as
+"liquidation" and is typically carried out as a way for the lender to recoup
+their losses in the event that the borrower is unable to fulfill their repayment
+obligations.
+
+During the liquidation process, the collateral tokens that have been pledged by
+the borrower as security for the loan are transferred from the borrower's
+account to the lender's account. This transfer is initiated by the lender and is
+typically triggered when the borrower fails to repay the loan by the agreed upon
+deadline. Once the collateral has been transferred, the lender can then sell it
+off in order to recoup their losses and compensate for the unpaid loan.
+
+## Keeper method
+
+```go title="x/loan/keeper/msg_server_liquidate_loan.go"
+package keeper
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	"loan/x/loan/types"
+)
+
+func (k msgServer) LiquidateLoan(goCtx context.Context, msg *types.MsgLiquidateLoan) (*types.MsgLiquidateLoanResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	loan, found := k.GetLoan(ctx, msg.Id)
+	if !found {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrKeyNotFound, fmt.Sprintf("key %d doesn't exist", msg.Id))
+	}
+	if loan.Lender != msg.Creator {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "Cannot liquidate: not the lender")
+	}
+	if loan.State != "approved" {
+		return nil, sdkerrors.Wrapf(types.ErrWrongLoanState, "%v", loan.State)
+	}
+	lender, _ := sdk.AccAddressFromBech32(loan.Lender)
+	collateral, _ := sdk.ParseCoinsNormalized(loan.Collateral)
+	deadline, err := strconv.ParseInt(loan.Deadline, 10, 64)
+	if err != nil {
+		panic(err)
```

Comment:
This is the first time we validate the `Deadline`, maybe it's a little too late :)

This is actually a valid exploit, how to borrow funds and never get liquidated ? You just need to request a loan with an invalid `Deadline` :smiling_imp:

Reply:
https://github.com/ignite/cli/pull/3352/commits/5821f7bf84f2667a904744b5e04ccaeb11a0d0d6